src/main/java/io/stargate/sgv2/jsonapi/util/JsonUtil.java

```diff
@@ -58,4 +61,22 @@ public static boolean equalsOrdered(JsonNode node1, JsonNode node2) {
     // For other nodes default equals() works fine:
     return Objects.equals(node1, node2);
   }
+
+  /**
+   * Helper method that will see if given {@link JsonNode} is an EJSON-encoded "Date" (aka
+   * Timestamp) value and if so, constructs and returns matching {@link java.util.Date} value. See
+   * {@href https://docs.meteor.com/api/ejson.html} for details on encoded value.
+   *
+   * @param json JSON value to check
+   * @return
+   */
+  public static Date extractEJsonDate(JsonNode json) {
+    if (json.isObject() && json.size() == 1) {
+      JsonNode value = json.path(EJSON_VALUE_KEY_DATE);
+      if (value.isIntegralNumber() && value.canConvertToLong()) {
+        return new Date(value.longValue());
+      }
+    }
+    return null;
```

Comment:
Do we need to throw error if the incoming value is not a number?

Reply:
Hmmmh. Good question. Yes, probably should, will add.

Reply:
@maheshrajamani Added better validation, now catches every JSON Object that looks like EJSON encoded (1 entry, key starts with "$").